solidity/test/BondedECDSAKeepFactoryTest.js

```diff
@@ -307,44 +309,39 @@ contract("BondedECDSAKeepFactory", async (accounts) => {
       )
     })
 
-    it("returns false if the operator stake is below minimum", async () => {
+    it("returns false if the operator stake dropped", async () => {
+      await stakeOperators(members, minimumStake.muln(50))
+
       await keepFactory.registerMemberCandidate(application, {
         from: members[0],
       })
-
-      await stakeOperators(members, minimumStake.subn(1))
+      await stakeOperators(members, minimumStake.muln(49))
 
       assert.isFalse(
         await keepFactory.isOperatorUpToDate(members[0], application)
       )
     })
 
-    it("returns true if the operator stake changed insufficiently", async () => {
+    it("returns false if the operator stake dropped below minimum", async () => {
       await keepFactory.registerMemberCandidate(application, {
         from: members[0],
       })
 
-      // We multiply minimumStake as sortition pools expect multiplies of the
-      // minimum stake to calculate stakers weight for eligibility.
-      // We subtract 1 to get the same staking weight which is calculated as
-      // `weight = floor(stakingBalance / minimumStake)`.
-      await stakeOperators(members, minimumStake.mul(new BN(2)).sub(new BN(1)))
+      await stakeOperators(members, minimumStake.subn(1))
 
-      assert.isTrue(
+      assert.isFalse(
         await keepFactory.isOperatorUpToDate(members[0], application)
       )
     })
 
-    it("returns false if the operator stake is above minimum", async () => {
+    it("returns false if the operator stake increased", async () => {
       await keepFactory.registerMemberCandidate(application, {
         from: members[0],
       })
 
-      // We multiply minimumStake as sortition pools expect multiplies of the
-      // minimum stake to calculate stakers weight for eligibility.
-      await stakeOperators(members, minimumStake.mul(new BN(2)))
+      await stakeOperators(members, minimumStake.addn(1))
 
-      assert.isFalse(
+      assert.isTrue(
```

Comment:
Could you please explain this change? Have some rules changed on the sortition pool side?

Reply:
I rethought tests after your comment and pushed more precise cases in https://github.com/keep-network/keep-ecdsa/pull/377/commits/3a61bf47809ab3b8fa01c27fc79473fa80934464

From https://github.com/keep-network/sortition-pools/pull/74:

> Minimum stake is no longer a hardcoded value and can change over time. To accommodate potential changes in minimum stake, we now use a constant pool weight divisor for bonded and standard sortition pool just like in the fully backed pool.
> 
> The initial minimum stake value is passed as a pool constructor parameter and the current value is later passed as an argument to select group function.

We agreed with @eth-r to set the pool weight divisor to 1 KEEP. There is an additional factor here: EVM performs implicit floor for `uint256` division. As a result:
- if stake drops by `10^-(18) KEEP`, `isOperatorUpToDate` returns `false`; this is covered by `returns false if the operator stake dropped` case
- if stake drops by `1 KEEP`, `isOperatorUpToDate` returns `false`; this is not covered by a separate case but is obvious from the previous one
- if stake increases by `1 KEEP`, `isOperatorUpToDate` returns `false`; this is covered by `returns false if the operator stake increased` case
- if stake increases by `10^(-18) KEEP`, `isOperatorUpToDate` returns `true`; this is covered by `returns true if the operator stake increase is below weight precision` case